### DISCUSSION TITLE
Share the layout data on serverMemo (Reactive layout)

### DISCUSSION THREAD

**Discussion by ph7jack** [livewire/livewire#5190] [3. Feature ideas]
Hello, it would be nice to have the layout data in the `serverMemo`, it'll allow listening to changes from the js and make the layout reactive.

I added a method to get the `preRenderedView` in the `Livewire\Component` class
![image](https://user-images.githubusercontent.com/38913462/185522400-a0895b44-8a64-4961-b713-38ccdd8032a4.png)

In the `LifecycleManager` class I've added some lines in the `dehydrate` method
![image](https://user-images.githubusercontent.com/38913462/185521023-70599291-1eb2-4f27-96bd-32945537369b.png)

Then I'll be able to listen to the component changes by the event `message.processed` and have access to the `layoutData`
![image](https://user-images.githubusercontent.com/38913462/185521258-1f79e9b1-ef0d-4f0d-8c1c-5e7140217052.png)

So I'll be able to always update the page title by just listening for one event instead of creating a big logic to dispatch events to perform this action.
I can create a PR with tests for this idea, I'm not sure where will be the best place to put this feature, but it will be really useful for building reactive layouts.

I looked deep and I'm also able to do the same behavior by listening from the event `component.dehydrate.subsequent` on the server side
Maybe add it as a class in the `src/Features` folder. 
![image](https://user-images.githubusercontent.com/38913462/185522846-698287b0-f552-4068-b564-536763dd7d91.png)

@calebporzio @joshhanley Any opinions?

**Comment by joshhanley** [comment]
@PH7-Jack I'm not sure if it's still happening or not, but I'd imagine if Livewire gets SPA mode support, then it would handle these sorts of things. 

I personally think it would be better to add some controls for updating basic things like `title` instead of sending through the whole layout (some people already complain about Eloquent model meta data).

I would probably hold off on the PR for now, as Caleb will be demoing V3 at Laracon in just over a week.
